ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -45,18 +46,75 @@ main = do
             success <- runIngredient
             unless success exitFailure
       where
+        bsss = [defaultInMemoryBSS, defaultLMDB_BSS]
+        ns   = [10_000, 20_000]
+
+        showBackingStoreSelector bss = case bss of
+          InMemoryBackingStore -> "inmem"
+          LMDBBackingStore _   -> "lmdb"
+
         benchmarkJustAddingTransactions =
-            bgroup "Just adding" $
-                fmap benchAddNTxs [10_000, 1_000_000]
+            bgroup "Just adding" [
+                bgroup "scenario1" [
+                    benchAddNTxs' bss n $ do
+                        theCandidateTransactionsHaveLinkedTxs Nothing n
+                  | bss <- bsss
+                  , n <- ns
+                  ]
+              , bgroup "scenario2 2160" [
+                    benchAddNTxs' bss n $ do
+                        clTxs <- theChangelogHasLinkedTxs 2_160
+                        theCandidateTransactionsHaveLinkedTxs (Just $ last clTxs) n
+                  | bss <- bsss
+                  , n <- ns
+                  ]
+              , bgroup "scenario3" [
+                    benchAddNTxs' bss n $ do
+                        theCandidateTransactionsHave  n
+                  | bss <- bsss
+                  , n <- ns
+                  ]
+              , bgroup "scenario4" [
+                    benchAddNTxs' bss n $ do
+                        bTxs <- theBackingStoreHas n
+                        theCandidateTransactionsConsume bTxs
+                  | bss <- bsss
+                  , n <- ns
+                  ]
+              , bgroup "scenario5" [
+                    benchAddNTxs' bss n $ do
+                        cTxs <- theChangelogHas n
+                        theCandidateTransactionsConsume cTxs
+                  | bss <- bsss
+                  , n <- ns
+                  ]
+              , bgroup "scenario6" [
+                    benchAddNTxs' bss n $ do
+                        bTxs <- theBackingStoreHas n
+                        cTxs <- theChangelogConsumes bTxs
+                        theCandidateTransactionsConsume cTxs
+                  | bss <- bsss
+                  , n <- ns
+                  ]
+              ]
           where
-            benchAddNTxs n =
-                env (let txs = mkNTryAddTxs n in fmap (, txs) (openMempoolWithCapacityFor txs))
+            benchAddNTxs' :: BackingStoreSelector IO -> Int -> ScBuilder () -> Benchmark
+            benchAddNTxs' bss n scenario =
+              let (params, txs) = fromScenario defaultLedgerDbCfg bss (build scenario)
+              in  benchAddNTxs params txs n
+
+            benchAddNTxs params txs0 n =
```

Comment:
If we don't need `benchAddNTxs` anywhere else, I'd say we rename  `benchAddNTxs'` to `benchAddNTxs` and use the RHS of this definition.